### ISSUE TITLE
Error in read_camtrapdp(), with some packages

### ISSUE THREAD

**Issue by ddachs** [inbo/camtrapdp#131]
With some packages I get an error, when using the read_camtrapdp() function: 
![image](https://github.com/user-attachments/assets/951f69d5-187a-4f2c-90bb-7039b6dd116b)

My camtrapDP output originates from TRAPPER, but it is not always the case with TRAPPER DPs. I cannot spot differences between working and failing packages. 

It seems, that the error happens in the frictionless:::read_from_path function. I tried executing read_camtrapdp function step by step, but eventually the purrr::map_df function worked just fine with paths as argument, wich is confusing to me. Maybe others have better luck, finding the problem.  

[example_DP.zip](https://github.com/user-attachments/files/17327134/example_DP.zip)

**Comment by peterdesmet** [comment]
The error is caused by extra columns in the CSV files in `example_DP.zip` that are not part of the Camtrap DP standard. Currently frictionless cannot (and according to the standard should not) deal with those.

The erroneous columns are:

- deployments.csv: ok
- media.csv: `_id` at the end
- observations.csv: `_id` at the end

TRAPPER should not add these columns (ping @kbubnicki). Note also that `metadata.json` should be named `datapackage.json`

---

We had a similar issue with Agouti, and I provided a temporary fix in R: https://gist.github.com/peterdesmet/fa9e6dfbd3020055d137d641492c47a3

For this case, you should do:

```R
# My files
directory <- "~/Downloads/example_DP"

# Load libraries
library(readr)
library(dplyr)

# Read, fix and write media
media <- read_csv(
  file.path(directory, "media.csv"),
  col_types = cols(.default = "c")
)
media <-
  media %>%
  dplyr::select(-`_id`)
readr::write_csv(media, file.path(directory, "media.csv"))

# Read, fix and write observations
obs <- read_csv(
  file.path(directory, "observations.csv"),
  col_types = cols(.default = "c")
)
obs <-
  obs %>%
  dplyr::select(-`_id`)
readr::write_csv(obs, file.path(directory, "observations.csv"))

# Read package
library(camtrapdp)
datapackage_path <- file.path(directory, "metadata.json")
package <- camtrapdp::read_camtrapdp(datapackage_path)
```

**Comment by ddachs** [comment]
Thanks for the quick response. That should be an easy fix on the Trapper side. Would it be an option, to raise a warning, if the camtrapDP is non standard form? I think we will see several entities formating their historical data into camtrapdp format by hand.  A check_camtrapdp() function would come in handy, to see if the job is done.

**Comment by peterdesmet** [comment]
> Would it be an option, to raise a warning, if the camtrapDP is non standard form?

Yes, I want that too, but the change has to happen in the underlaying [frictionless](https://docs.ropensci.org/frictionless/) dependency, which we also maintain. It is a change that is part of a wider batch of changes that we have to find the time for.

> A check_camtrapdp() function would come in handy, to see if the job is done.

Yes, there is in fact already a function with that name, that is called by every function to do some basic checks on the provided `x`. It works on an already-read package (`x`) though and the error you describe blocks getting to that stage. Longer term, I want a `validate_camtrapdp()` that does a number of checks: not only expected columns, but also e.g. missing coordinates, etc. It wouldn't be called by every function, but by the user.
